packages/apex-node/src/tests/codeCoverage.ts

```diff
@@ -117,56 +115,83 @@ export class CodeCoverage {
-          type: item.ApexClassOrTrigger.Id.startsWith('01p')
-            ? 'ApexClass'
-            : 'ApexTrigger',
-          numLinesCovered: item.NumLinesCovered,
-          numLinesUncovered: item.NumLinesUncovered,
-          percentage,
-          coveredLines: item.Coverage.coveredLines,
-          uncoveredLines: item.Coverage.uncoveredLines
-        };
-      }
-    );
+    const totalCodeCoverageResults: CodeCoverageResult[] = [];
+
+    codeCoverageAggregates.forEach(chunk => {
+      const codeCoverageResults: CodeCoverageResult[] = chunk.records.map(
+        item => {
+          totalLinesCovered += item.NumLinesCovered;
+          totalLinesUncovered += item.NumLinesUncovered;
+          const totalLines = item.NumLinesCovered + item.NumLinesUncovered;
+          const percentage = calculatePercentage(
+            item.NumLinesCovered,
+            totalLines
+          );
+
+          return {
+            apexId: item.ApexClassOrTrigger.Id,
+            name: item.ApexClassOrTrigger.Name,
+            type: item.ApexClassOrTrigger.Id.startsWith('01p')
+              ? 'ApexClass'
+              : 'ApexTrigger',
+            numLinesCovered: item.NumLinesCovered,
+            numLinesUncovered: item.NumLinesUncovered,
+            percentage,
+            coveredLines: item.Coverage.coveredLines,
+            uncoveredLines: item.Coverage.uncoveredLines
+          };
+        }
+      );
+
+      totalCodeCoverageResults.push(...codeCoverageResults);
+    });
 
     return {
-      codeCoverageResults,
+      codeCoverageResults: totalCodeCoverageResults,
       totalLines: totalLinesCovered + totalLinesUncovered,
       coveredLines: totalLinesCovered
     };
   }
 
   private async queryAggregateCodeCov(
     apexClassIdSet: Set<string>
-  ): Promise<ApexCodeCoverageAggregate> {
-    let str = '';
-    apexClassIdSet.forEach(elem => {
-      str += `'${elem}',`;
-    });
-    str = str.slice(0, -1);
-
+  ): Promise<ApexCodeCoverageAggregate[]> {
     const codeCoverageQuery =
       'SELECT ApexClassOrTrigger.Id, ApexClassOrTrigger.Name, NumLinesCovered, NumLinesUncovered, Coverage FROM ApexCodeCoverageAggregate WHERE ApexClassorTriggerId IN (%s)';
-    const codeCoverageResuls = (await this.connection.tooling.autoFetchQuery(
-      util.format(codeCoverageQuery, `${str}`)
-    )) as ApexCodeCoverageAggregate;
-    return codeCoverageResuls;
+    return this.fetchResults(apexClassIdSet, codeCoverageQuery);
+  }
+
+  private async fetchResults<
+    T extends ApexCodeCoverage | ApexCodeCoverageAggregate
+  >(idSet: Set<string>, selectQuery: string): Promise<T[]> {
+    const queries = this.createQueries(selectQuery, idSet);
+
+    const queryPromises = queries.map(query => {
+      return this.connection.tooling.autoFetchQuery(query) as Promise<T>;
+    });
+
+    return await Promise.all(queryPromises);
+  }
+
+  private createQueries(selectQuery: string, idSet: Set<string>): string[] {
```

Comment:
After making the change in asyncTests.ts and getting the calls to HTTP with large record sets to pass, I discovered that we then had an issue downstream, and that getting large record sets of ApexCodeCoverage and ApexCodeCoverageAggregate also were failing, and I applied a similar fix as the one in asyncTests.ts.

Reply:
Great catch jeff! Were they failing due to the same reason, due to the query being too large?

Reply:
Yes - once the query in asyncTests was working, these other queries processed larger record sets, and they were then failing b/c of the same reason.